src/SupportView.ts

```diff
@@ -119,6 +119,9 @@ export default class SupportView extends EventHandler {
     this.datasets = convertTableToVectors(await listData())
       .filter((d) => d.idtypes.indexOf(this.idType) >= 0 && isPossibleDataset(d));
 
+    const matrixColumns = await Promise.all(this.datasets.filter((d) => d.desc.type === 'matrix').map(splitMatrixInVectors));
+    this.datasets.push(...[].concat(...matrixColumns));
```

Comment:
`matrixColumns` has the type `IDataType[][]` using the first concat I create a flat array out of it `IDataType[]` which is later on pushed to `this.datasets`